tests/unit/s2n_nst_early_data_indication_test.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "tls/extensions/s2n_early_data_indication.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test s2n_nst_early_data_indication_should_send */
+    {
+        /* Safety check */
+        EXPECT_FALSE(s2n_nst_early_data_indication_extension.should_send(NULL));
+
+        struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+        EXPECT_NOT_NULL(conn);
+
+        /* Should not send if max_early_data_size not set */
+        EXPECT_FALSE(s2n_nst_early_data_indication_extension.should_send(conn));
+
+        /* Should not send if max_early_data_size set to 0 */
+        EXPECT_SUCCESS(s2n_connection_set_server_max_early_data_size(conn, 0));
+        EXPECT_FALSE(s2n_nst_early_data_indication_extension.should_send(conn));
+
+        /* Should send if max_early_data_size set to non-zero */
+        uint32_t server_max = 13;
+        EXPECT_SUCCESS(s2n_connection_set_server_max_early_data_size(conn, server_max));
+        EXPECT_TRUE(s2n_nst_early_data_indication_extension.should_send(conn));
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test s2n_nst_early_data_indiction_send */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+        EXPECT_NOT_NULL(conn);
+        conn->actual_protocol_version = S2N_TLS13;
+
+        struct s2n_stuffer output = { 0 };
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&output, 0));
+
+        const uint32_t expected_max_early_data_size = 13;
```

Comment:
Are these sizes just randomly chosen? Any additional value to running the check in a loop of possible values? I suppose that is more of a cbmc thing, just wondering about the sizes.

Reply:
Yes, just randomly chosen. The value doesn't really matter, since we just assign it. I do often write tests that loop over possible values, but only when the values actually matter in some way. I don't think it would provide any value here.

Reply:
Basically, testing a=b for uint32_max values seems a little excessive xD